x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/blobstore/cache/SearchableSnapshotsBlobStoreCacheIntegTests.java

```diff
@@ -205,33 +250,36 @@ public void testBlobStoreCache() throws Exception {
 
         assertAcked(client().admin().indices().prepareDelete(restoredIndex));
 
-        logger.info("--> mount snapshot [{}] as an index for the second time", snapshot);
-        final String restoredAgainIndex = mountSnapshot(
+        cacheEnabled = randomBoolean();
```

Comment:
The second time the index is mounted can now be fully randomized between full cache/partial cache/no cache.